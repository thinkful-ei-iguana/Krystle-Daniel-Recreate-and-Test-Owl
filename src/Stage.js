import React from 'react';
import StagedParticipant from './StagedParticipant';

function Stage(props) {
  const staged = props.participants.filter(participant => {
    return participant.onStage
  }).map(participant => {
    return <StagedParticipant name={participant.name} />
  });


  return (
    <div className="stage-area">
      {staged}
    </div>
  );
} 

export default Stage;