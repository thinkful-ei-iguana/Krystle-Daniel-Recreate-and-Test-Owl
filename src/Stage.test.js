import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from './Stage';


describe('Stage', () => {

  const participants = [
    {
        id: 1,
        name: 'Koren Templeton',
        avatar:
            'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
    },
    {
        id: 2,
        name: 'Caty Flucker',
        avatar:
            'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: false
    }
  ]

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Stage participants={participants}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Stage participants={participants}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

 
});