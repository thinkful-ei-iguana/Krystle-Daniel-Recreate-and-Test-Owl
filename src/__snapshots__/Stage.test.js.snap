// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage renders the UI as expected 1`] = `
<div
  className="stage-area"
>
  <div
    className="staged-participant"
  >
    Koren Templeton
  </div>
</div>
`;
